cloudflare/resource_cloudflare_waf_group.go

```diff
@@ -0,0 +1,210 @@
+package cloudflare
+
+import (
+	"fmt"
+	"strings"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCloudflareWAFGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareWAFGroupCreate,
+		Read:   resourceCloudflareWAFGroupRead,
+		Update: resourceCloudflareWAFGroupUpdate,
+		Delete: resourceCloudflareWAFGroupDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareWAFGroupImport,
+		},
+
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"package_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "on",
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					switch value := val.(string); value {
+					case "on", "off":
+						break
+					default:
+						errs = append(errs, fmt.Errorf("%q must be either 'on' or 'off', got: %s", key, value))
+					}
+					return
+				},
```

Comment:
There is a built in helper for this, [`validation.StringInSlice()`](https://github.com/hashicorp/terraform/blob/00a76d019a3e5b4d9d040a3e28c9e0b4e56ad951/helper/validation/validation.go#L127-L147) which simplifies the reading of this validation.

Reply:
Updated!